### PR TITLE
Updated classes and code from Nexmo to Vonage

### PR DESCRIPTION
**Pull request by dragonmantank**
This handles the rebrand of the library from Nexmo to Vonage. Functionally it is equivalent to `nexmo/client-core:2.2.0` 

* Namespace switch from `\Nexmo` to `\Vonage`
* All unit tests and source code updated to the new namespacing
* Composer package name updated to `vonage/client-core`
* License file link in page docblock will work once the `vonage-php` repo is updated

### CODE REVIEW

File: src/Verify/Verification.php [hunk 1]
```diff
@@ -187,10 +187,10 @@ public function isDirty()
      * If do not set number in international format or you are not sure if number is correctly formatted, set country
      * with the two-character country code. For example, GB, US. Verify works out the international phone number for
      * you.
-     * @link https://docs.nexmo.com/verify/api-reference/api-reference#vrequest
+     * @link https://docs.Vonage.com/verify/api-reference/api-reference#vrequest
```
Comment (GregHolmes) [inline src/Verify/Verification.php:190]: This link won't work. Should this remain `docs.nexmo.com`?

There's a few of these

File: test/Account/responses/secret-management/create-validation.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "type": "https://developer.nexmo.com/api-errors/account/secret-management#validation",
+  "type": "https://developer.Vonage.com/api-errors/account/secret-management#validation",
```
Comment (GregHolmes) [inline test/Account/responses/secret-management/create-validation.json:2]: These links won't work either.
Should these remain as `developer.nexmo.com`?

File: test/Application/ClientTest.php [hunk 3]
```diff
@@ -45,11 +45,11 @@ class ClientTest extends TestCase
 
     public function setUp()
     {
-        $this->nexmoClient = $this->prophesize('Nexmo\Client');
-        $this->nexmoClient->getApiUrl()->willReturn('http://api.nexmo.com');
+        $this->VonageClient = $this->prophesize('Vonage\Client');
```
Comment (GregHolmes) [inline test/Application/ClientTest.php:48]: Just me being pedantic, but should this be lower case `v` for `$this->vonageClient`?

**Review by GregHolmes — CHANGES_REQUESTED** [review]
I've made a couple of comments of some issues I've found scrolling through the PR. Haven't tested the functionality though.

File: test/Application/ClientTest.php (hunk 3 above)
Comment (dragonmantank) [reply test/Application/ClientTest.php]: Nope, should be lowercase. Corrected

File: test/Account/responses/secret-management/create-validation.json (hunk 2 above)
Comment (dragonmantank) [reply test/Account/responses/secret-management/create-validation.json]: Fixed

File: src/Verify/Verification.php (hunk 1 above)
Comment (dragonmantank) [reply src/Verify/Verification.php]: Fixed, will also make sure that domain is on our radar

File: src/Client/Response/AbstractResponse.php [hunk 4]
```diff
@@ -1,12 +1,12 @@
 <?php
 /**
- * Nexmo Client Library for PHP
+ * Vonage Client Library for PHP
  *
- * @copyright Copyright (c) 2016 Nexmo, Inc. (http://nexmo.com)
- * @license   https://github.com/Nexmo/nexmo-php/blob/master/LICENSE.txt MIT License
+ * @copyright Copyright (c) 2016 Vonage, Inc. (http://vonage.com)
+ * @license   https://github.com/vonage/vonage-php/blob/master/LICENSE.txt MIT License
```
[the remark this answers is not in the document]
Comment (dragonmantank) [reply src/Client/Response/AbstractResponse.php]: Corrected

**Review by GregHolmes — APPROVED** [review]

**Pull request merged**